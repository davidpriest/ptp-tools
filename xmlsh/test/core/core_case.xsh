# core_case.xsh
# Test of case

echo test1
case foo in 
	foo) echo success ;;
	bar) echo failure ; exit 1 ;;
esac

echo test2
case foo in 
	ooo) echo failure ; exit 1;;
	f*)	 echo success ;;
esac

echo test3
case foo in 
	?oo|ooo) echo success ;;	# should hit ?oo first 
	f*)	 echo failure ; exit 1 ;;
	
esac

# Test default and empty commands
case foo in 
   foo) ;; # Nothing
   *) echo Failure wrong default ; exit 1 ;;
esac

case foo in 
   oo) ;; # Nothing
   *) ;; 
esac



# Test IO to case
[ -d $TMPDIR/_xmlsh ] || mkdir $TMPDIR/_xmlsh
T1=$TMPDIR/_xmlsh/case1.tmp
T2=$TMPDIR/_xmlsh/case2.tmp

echo test4

echo data > $T1

case foo in 
	foo) read d ; echo data is $d ;;
	bar) echo failure ; exit 1 ;;
esac < $T1 > $T2

cat $T2

rm $T1 $T2
