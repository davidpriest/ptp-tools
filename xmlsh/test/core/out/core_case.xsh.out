test1
success
test2
success
test3
success
test4
data is data
